### PR TITLE
Implement the gRPC requests signing process

### PR DESCRIPTION
**Pull request by supl**
This PR implements the process of signing the gRPC requests on the client side.
The implementation follows
https://github.com/scalar-labs/scalar/blob/master/client/src/main/java/com/scalar/dl/client/util/RequestSigner.java

The implementation is based on and extends gRPC-generated files (the rpc package)

### CODE REVIEW

File: rpc/request_sign_with.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package rpc
```
Comment (feeblefakie) [inline rpc/request_sign_with.go]: The file name should be something like `request_signer.go`?
I feel it's a bit weird to see a program file ending with prepositions.

**Review by feeblefakie — COMMENTED** [review]
Overall looking good. Left a minor comment for a naming.

File: rpc/request_sign_with.go (hunk 1 above)
Comment (supl) [reply rpc/request_sign_with.go]: Understood.
Sorry for my bad naming.
(it was requset_signer.go in the beginning)

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**